src/charm_state.py

```diff
@@ -341,15 +342,52 @@ def check_fields(cls, values: dict) -> dict:
         return values
 
 
+LTS_IMAGE_VERSION_TAG_MAP = {"22.04": "jammy", "24.04": "noble"}
+
+
+class BaseImage(str, Enum):
+    """The ubuntu OS base image to build and deploy runners on."""
+
+    JAMMY = "jammy"
+    NOBLE = "noble"
+
+    def __str__(self) -> str:
+        """Interpolate to string value."""
+        return self.value
+
+    @classmethod
+    def from_charm(cls, charm: CharmBase) -> "BaseImage":
+        """Retrieve the base image tag from charm.
+
+        Args:
+            charm: The charm instance.
+
+        Raises:
+            ValueError if an unsupporte base image is passed in.
+
+        Returns:
+            The base image configuration of the charm.
+        """
+        image_name = charm.config.get(BASE_IMAGE_CONFIG_NAME, "jammy").lower().strip()
+        try:
+            return cls(image_name)
+        except ValueError as exc:
```

Comment:
Perhaps we just do a get from `LTS_IMAGE_VERSION_TAG_MAP` and put the current value as the default?

Reply:
Fixed, thank you!